src/main/java/org/breedinginsight/brapi/v2/services/BrAPITrialService.java

```diff
@@ -17,19 +18,38 @@
 @Singleton
 public class BrAPITrialService {
 
-    @Property(name = "brapi.server.reference-source")
-    private String referenceSource;
-
     private final BrAPITrialDAO trialDAO;
-    private final ProgramService programService;
+    private final BrAPIObservationUnitDAO ouDAO;
 
     @Inject
-    public BrAPITrialService(ProgramService programService, BrAPITrialDAO trialDAO) {
-        this.programService = programService;
+    public BrAPITrialService( BrAPITrialDAO trialDAO,BrAPIObservationUnitDAO ouDAO) {
         this.trialDAO = trialDAO;
+        this.ouDAO = ouDAO;
     }
 
     public List<BrAPITrial> getExperiments(UUID programId) throws ApiException, DoesNotExistException {
         return trialDAO.getTrials(programId);
     }
+
+    public BrAPITrial getTrialDataByUUID(UUID programId, UUID trialId, boolean stats) throws DoesNotExistException {
+        try {
+            BrAPITrial trial = trialDAO.getTrialById(programId,trialId).orElseThrow(() -> new DoesNotExistException("Trial does not exist"));
+            //Remove the [program key] from the trial name
+            trial.setTrialName( Utilities.removeUnknownProgramKey( trial.getTrialName()) );
+            if( stats ){
+                int environmentsCount = 1; // For now this is hardcoded to 1, because we are only supporting one environment per experiment
+                long germplasmCount = countGermplasm(programId, trial.getTrialDbId());
+                trial.putAdditionalInfoItem("environmentsCount", environmentsCount);
+                trial.putAdditionalInfoItem("germplasmCount", germplasmCount);
+            }
+            return trial;
+        } catch (ApiException e) {
+            throw new InternalServerException(e.getMessage(), e);
```

Comment:
Could you add a error log message here.  You can use `Utilities#generateApiExceptionLogMessage` to generate a log message from the `ApiException` object.